internal/db/schema/migrations/oss/postgres/99/02_worker_controller_tables.up.sql

```diff
@@ -83,6 +83,27 @@ create table server_worker (
 comment on table server_worker  is
   'server_worker is a table where each row represents a Boundary worker.';
 
+create or replace function
+  validate_key_id_on_insert()
+  returns trigger
+as $$
+begin
+    if new.key_id is not null then
+      perform from
+        worker_auth_authorized
+      where
+        worker_key_identifier= new.key_id;
+    if not found then
+      raise 'key_id %s does not match a worker auth record', new.key_id;
+    end if;
+  end if;
+  return new;
+end;
+$$ language plpgsql;
+
+create trigger validate_key_id_on_insert before insert on server_worker
+  for each row execute procedure validate_key_id_on_insert();
+
```

Comment:
I think, key_id is immutable and perhaps should be added to the immutable_columns func call for server_worker.  If this is true, then we should add an "update" test for this as well.

Reply:
Key_id can change when the worker rotates its creds